silver/views.py

```diff
@@ -43,6 +43,26 @@ def invoice_pdf(request, invoice_id):
 
 
 @csrf_exempt
+def initialize_transaction(request, transaction_uuid):
+    try:
+        uuid = UUID(transaction_uuid, version=4)
+    except ValueError:
+        raise Http404
+
+    transaction = get_object_or_404(Transaction, uuid=uuid)
+
+    transaction.last_access = timezone.now()
+    transaction.save()
+
+    if transaction.payment_processor.was_transaction_initialized(transaction,
```

Comment:
And here you'd only do `handle_transaction_response(transaction, request)` and based on the `transaction.status` you redirect to the proper url. This way we can have a `canceled_url` too, which is pretty common to have.